### PR TITLE
create example-react-hooks-useContext

### PR DESCRIPTION
**Pull request by iqbal125**
simple test example to test if a child component can update the context state.

### CODE REVIEW

**Comment by alexkrolick** [comment]
One comment - naming the inner component `TestComponent` could make people think that is a special for component just for the test instead of part of the App. Can we change that and add a note along the lines of:

> The basic idea is to test your components as they are used. Instead of testing the context-providing component separately, your main focus should be testing that the features connected to the context work correctly.

File: docs/example-react-hooks-useContext.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
+This is a simple example to test a component using the useContext hook. 
+
+
+The context object
+
+```
+import React from 'react';
```
Comment (afontcu) [inline docs/example-react-hooks-useContext.md]: Hi! Some parts of the code are using semicolons, and some others aren't. I just checked and I'd say all demo code from the docs are written without semicolons, so I'd suggest removing them from the examples too!

Comment (alexkrolick) [reply docs/example-react-hooks-useContext.md]: If you pull down the repo locally the git hooks should format the docs on save

File: docs/example-react-hooks-useContext.md [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+import App from '../../../App'
+
+import Context from '../../store/context';
+
+afterEach(cleanup)
+
+it('Context value is updated by child component', () => {
+
+   const { container, getByText } = render(<App>
+                                            <Context.Provider>
```
Comment (eps1lon) [inline docs/example-react-hooks-useContext.md]: Missing a value here. I'm not familiar with this usage. Does this set the value to `undefined` or the default context value? If it sets it to `undefined` we should make it explicit. If it uses the default value I would rather omit the provider component.

Comment (alexkrolick) [reply docs/example-react-hooks-useContext.md]: The default value is only used when there is no Provider at all above a Consumer

**Comment by iqbal125** [comment]
@kentcdodds  @eps1lon  @alexkrolick  @afontcu Made changes based on feedback let me know what you think.

File: docs/example-react-hooks-useContext.md [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+
+const App = () => {
+  const [state, setState] = useState("Some Text")
+
+  const changeText = () => {
+    setState("Some Other Text")
+  }
+
+   <h1> Basic Hook useContext</h1>
```
Comment (afontcu) [inline docs/example-react-hooks-useContext.md]: isn't a `return` statement missing here? I'd run all code in docs, to make sure it is working properly. We don't want to provide broken demos to our readers! 😉

File: docs/example-react-hooks-useContext.md [hunk 4]
```diff
@@ -0,0 +1,95 @@
+This is a simple example to test a component using the useContext hook. 
+
+
+The context object
+
+```
```
Comment (afontcu) [inline docs/example-react-hooks-useContext.md]: You should add the language type for the syntax highlighter: ` ```js ` or ` ```jsx ` or similar

File: docs/example-react-hooks-useContext.md (hunk 3 above)
Comment (iqbal125) [reply docs/example-react-hooks-useContext.md]: Wow your right cant believe I missed something so obvious

File: docs/example-react-hooks-useContext.md (hunk 2 above)
Comment (iqbal125) [reply docs/example-react-hooks-useContext.md]: I included it because I thought it would make the code easier to read, I'll change to render(<App/>)

**Comment by iqbal125** [comment]
@afontcu Made changes based on feedback let me know what you think.

File: docs/example-react-hooks-useContext.md [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+
+```jsx
+import React, { useContext } from 'react'
+
+import Context from '../context'
+
+const ChildComponent = () => {
+  const context = useContext(Context)
+
+  return (
```
Comment (afontcu) [inline docs/example-react-hooks-useContext.md]: Looks better! Thank you 🙌  could you please fix the indentation of the JSX elements?

```html
<div>
  <button ...>
    Change text
  </button>
  <p>...</p>
</div>
```

File: docs/example-react-hooks-useContext.md [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+
+
+const App = () => {
+  const [state, setState] = useState("Some Text")
+
+  const changeText = () => {
+    setState("Some Other Text")
+  }
+
+return (
```
Comment (afontcu) [inline docs/example-react-hooks-useContext.md]:
```suggestion
  return (
```

File: docs/example-react-hooks-useContext.md [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+
+the root parent component
+
+```jsx
+import React, { useState } from 'react'
+import Context from './context'
+
+
+const App = () => {
+  const [state, setState] = useState("Some Text")
```
Comment (afontcu) [inline docs/example-react-hooks-useContext.md]:
```suggestion
  const [state, setState] = useState('Some Text')
```

File: docs/example-react-hooks-useContext.md [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+import {render, fireEvent, cleanup} from '@testing-library/react'
+import App from './App'
+
+import Context from './context'
+
+afterEach(cleanup)
+
+it('Context value is updated by child component', () => {
+
+   const { container, getByText } = render(<App />);
```
Comment (afontcu) [inline docs/example-react-hooks-useContext.md]:
```suggestion
   const { getByText } = render(<App />)
```

File: docs/example-react-hooks-useContext.md [hunk 9]
```diff
@@ -0,0 +1,92 @@
+This is a simple example to test a component using the useContext hook. 
+
+
+The context object
```
Comment (afontcu) [inline docs/example-react-hooks-useContext.md]: I just noticed that other examples add the filename on top of the code snippet block, as in

```js
// context.js

import React from 'react'

....
```

I feel that could be seen as the "title" of the snippet. It'd be great to keep that strategy, to make things consistent. What do you think about it?

Comment (iqbal125) [reply docs/example-react-hooks-useContext.md]: yeah makes sense

**Comment by iqbal125** [comment]
@afontcu was able to make the changes let me know what you think.

File: docs/example-react-hooks-useContext.md [hunk 10]
```diff
@@ -0,0 +1,96 @@
...
+import {render, fireEvent, cleanup} from '@testing-library/react'
+import App from './App'
+
+import Context from './context'
+
+afterEach(cleanup)
+
+it('Context value is updated by child component', () => {
+
+   const { getByText } = render(<App />);
```
Comment (afontcu) [inline docs/example-react-hooks-useContext.md]:
```suggestion
   const { getByText } = render(<App />)
```

File: docs/example-react-hooks-useContext.md [hunk 11]
```diff
@@ -0,0 +1,96 @@
...
+
+  return (
+   <h1> Basic Hook useContext</h1>
+     <Context.Provider value={{changeTextProp: changeText,
+                               stateProp: state
+                                 }} >
+        <ChildComponent />
+     </Context.Provider>
+    </div>
+  );
```
Comment (afontcu) [inline docs/example-react-hooks-useContext.md]:
```suggestion
  )
```

**Comment by afontcu** [comment]
I've cleaned up the file a bit (whitespacing, naming, etc), but I'd love someone else to check that the _actual_ example test makes sense (from a React/useContext point of view). Not sure if using it with `useState` makes the example a bit cumbersome.

Your call! :)

**Review by alexkrolick — COMMENTED** [review]
It would be good to link this page to the `wrapper` docs to show how to deal with Context providers when you aren't rendering a whole app.

**Pull request closed without merging**